turbomoleio/output/data.py

```diff
@@ -1484,3 +1500,94 @@ def from_parser(cls, parser):
+            shortest_interatomic_distance: Shortest interatomic distance in the
+                system (in Angstroms).
+            direct_space_vectors: Lattice vectors of the system (in Angstroms).
+                The first vector will always be aligned with the x cartesian direction.
+                For 2D and 3D, the second vector is always in the xy cartesian plane.
+            reciprocal_space_vectors: Reciprocal lattice vectors of the system (in Angstroms^-1).
+                The physics definition of the reciprocal lattice vectors is used here, i.e.
+                    b1 = 2*pi/V * (a2 x a3)
+                    b2 = 2*pi/V * (a3 x a1)
+                    b3 = 2*pi/V * (a1 x a2)
+                The crystallographer's definition is easily recovered as:
+                    b'1 = b1 / (2*pi)
+                    b'2 = b2 / (2*pi)
+                    b'3 = b3 / (2*pi)
+
+        Returns:
+            PeriodicityData.
+
+        """
+        self.periodicity = periodicity
+        self.lattice_params = lattice_params
+        self.shortest_interatomic_distance = shortest_interatomic_distance
+        self.direct_space_vectors = direct_space_vectors
+        self.reciprocal_space_vectors = reciprocal_space_vectors
+
+    @classmethod
+    def from_parser(cls, parser):
+        """
+        Generates an instance of PeriodicityData from a parser based on the stdout
+        of a Turbomole executable. Returns None if no data could be parsed.
+
+        Args:
+            parser (Parser): the parser to be used to extract the data.
+
+        Returns:
+            PeriodicityData.
+        """
+        data = parser.periodicity_data
+        if not data:
+            return None
+
+        periodicity = data["periodicity"]
+        if periodicity == 1:
+            # a
+            lattice_params = [bohr_to_ang * data["tm_lattice_params"][0]]
+        elif periodicity == 2:
+            # a, b, gamma
+            lattice_params = [bohr_to_ang * data["tm_lattice_params"][0],
+                              bohr_to_ang * data["tm_lattice_params"][1],
+                              data["tm_lattice_params"][2]]
+        elif periodicity == 3:
+            # a, b, c, alpha, beta, gamma
+            lattice_params = [bohr_to_ang * data["tm_lattice_params"][0],
+                              bohr_to_ang * data["tm_lattice_params"][1],
+                              bohr_to_ang * data["tm_lattice_params"][2],
+                              data["tm_lattice_params"][3],
+                              data["tm_lattice_params"][4],
+                              data["tm_lattice_params"][5]]
+        else:
+            raise RuntimeError('"periodicity" should be 1, 2 or 3.')
+
+        direct_space_vectors = [[bohr_to_ang * xx for xx in vect]
+                                for vect in data["direct_space_vectors"]]
+        reciprocal_space_vectors = [[xx / bohr_to_ang for xx in vect]
```

Comment:
are the reciprocal lattice vectors worth storing?

Reply:
No idea, maybe not. In any case it is not very big. For 3d we can directly access it from the Lattice object. For 1 and 2D, you need to take only the 1x1 or 2x2 part of the reciprocal lattice. What do you think ? Should we remove it from the PeriodicityData object ?